File: web/hospital-web/src/api/servicios.ts
import { http } from "./httpClient";

export interface ServicioDto {
  idServicio: number;
  descripcion: string;
  tipo: string;
  precio: number;
  stock: number | null; 
}

export async function listarServicios() {
  const res = await http.get<ServicioDto[]>("/api/Servicios");
  return res.data;
}

export async function actualizarStockServicio(id: number, nuevoStock: number) {
  const res = await http.put(`/api/Servicios/${id}/stock`, {
    stock: nuevoStock
  });
  return res.data;
}

File: api/Hospital.Api/Dtos/Farmacia/Servicios.cs
namespace Hospital.Api.Dtos.Farmacia;

public class ServicioDto
{
    public int IdServicio { get; set; }
    public string Descripcion { get; set; } = string.Empty;
    public string Tipo { get; set; } = string.Empty;
    public decimal Precio { get; set; }
    public int? Stock { get; set; } 
}